core/src/main/java/org/bptlab/cepta/patterns/StaysInStationPattern.java

```diff
@@ -0,0 +1,67 @@
+package org.bptlab.cepta.patterns;
+
+import org.apache.flink.streaming.api.windowing.time.Time;
+import org.apache.flink.streaming.api.datastream.DataStreamUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.cep.PatternStream;
+import org.apache.flink.cep.CEP;
+import org.apache.flink.cep.pattern.*;
+import org.apache.flink.cep.pattern.conditions.*;
+import org.bptlab.cepta.models.events.train.LiveTrainDataProtos.LiveTrainData;
+import org.bptlab.cepta.models.events.correlatedEvents.StaysInStationEventProtos.StaysInStationEvent;
+
+import java.util.*;
+
+
+public class StaysInStationPattern {   
+    public static final Pattern<LiveTrainData, ?> staysInStationPattern = 
```

Comment:
this doesn't really test, if a train stays in station, i think. That's because it only tests, if a status=3 event is followed by a status!=4 event. It does not consider the train itself. so it could be events of a train A with status 3 followed by a train B with status 3.

If this behavior is intentional, please rename the pattern.

Reply:
we added a check for them to have the same trainId